heaps/min_heap.py

```diff
@@ -57,19 +77,48 @@ def heap_up(self, index):
             property is reestablished.
             
             This could be **very** helpful for the add method.
-            Time complexity: ?
-            Space complexity: ?
+            Time complexity: O(log n)
+            Space complexity: O(1)
         """
-        pass
+        if index == 0:
+            return
+        if self.store[index].key < self.store[(index-1) // 2].key:
+            self.swap(index, (index-1) // 2)
+            self.heap_up((index-1) // 2)
```

Comment:
I'd recommend inserting a variable to hold the parent index. To me, this is a bit hard to read. It's not immediately apparent to me what `(index-1) // 2` represents.
```suggestion
        parent_index = (index-1) // 2
        if self.store[index].key < self.store[parent_index].key:
            self.swap(index, parent_index)
            self.heap_up(parent_index)
```